File: tests/test2.py
# template file
import json
from typing import List


class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


def createLinkedList(nums):
    head = None
    for num in reversed(nums):
        newNode = ListNode(num, head)
        head = newNode
    
    return head


def linkedListToVector(node):
    intList = []
    while (node != None):
        intList.append(node.val)
        node = node.next
    
    return intList


# insert Solution class here
class Solution:
    def addTwoNumbers(self, l1, l2):
        dummy = cur = ListNode(0)
        carry = 0
        while l1 or l2 or carry:
            if l1:
                carry += l1.val
                l1 = l1.next
            if l2:
                carry += l2.val
                l2 = l2.next
            cur.next = ListNode(carry%10)
            cur = cur.next
            carry //= 10
        return dummy.next


with open("tc2.json", "r") as read_file:
    data = json.load(read_file)

sol = Solution()
isOk = True

listL1 = data["input"]["l1"]
listL2 = data["input"]["l2"]

for j, nums in enumerate(listL1):
    l1 = createLinkedList(listL1[j])
    l2 = createLinkedList(listL2[j])

    solutionNode = sol.addTwoNumbers(l1, l2)
    solutionList = linkedListToVector(solutionNode)
    
    if solutionList != data["expected"][j]:
        isOk = False
        break

if isOk:
    # write json to file
    output = {
        "result": "OK"
    }
    json_object = json.dumps(output, indent=4)
    print(json_object)

print("test completed")

File: tests/test2.js
// template file
const tc = require("./tc2.json")

class ListNode {
    constructor(val, next) {
        this.val = (val===undefined ? 0 : val)
        this.next = (next===undefined ? null : next)
    }
}

function createLinkedList(nums) {
    let head = null
    nums.forEach(num => {
        let newNode = new ListNode(num, head)
        head = newNode
    });

    return head
}

function linkedListToVector(node) {
    let intList = []
    while (node != null) {
        intList.push(node.val)
        node = node.next
    }

    return intList
}

// insert Solution class here
var addTwoNumbers = function(l1, l2) {
    let values = [];
    let overflow = 0;
    while (l1.next || l2.next) {
        const value = (l1.val + l2.val + overflow) % 10;
        overflow = l1.val + l2.val + overflow >= 10 ? 1 : 0;
        values.push(value);

        l1 = l1.next || new ListNode(0);
        l2 = l2.next || new ListNode(0);
    }
    const value = (l1.val + l2.val + overflow) % 10;
    overflow = l1.val + l2.val + overflow >= 10 ? 1 : 0;
    values.push(value);
    if (overflow === 1) {
        values.push(1);
    }
    let node = undefined;
    values.reverse().forEach(item => {
        if (node === undefined) {
            node = new ListNode(item);
        } else {
            const newNode = new ListNode(item);
            newNode.next = node;
            node = newNode;
        }
    });
    return node;
};

let isOk = true
const listL1 = tc.input.l1
const listL2 = tc.input.l2
for (let i = 0; i < listL1.length; i++) {
    const l1 = createLinkedList(listL1[i])
    const l2 = createLinkedList(listL2[i])

    const solutionNode = addTwoNumbers(l1, l2)
    const solutionList = linkedListToVector(solutionNode)

    for (let j = 0; j < solutionList.length; j++) {
        if (solutionList[j] != tc.expected[i][j]) {
            isOk = false
            break
        }   
    }
}

if (isOk) {
    const output = {
        "result": "OK"
    }
    const data = JSON.stringify(output)
    console.log(data)
}

console.log("test completed")